integration/kubernetes/helm/alluxio/templates/alluxio-master.yaml

```diff
@@ -23,124 +35,122 @@ spec:
-              cpu: {{ .Values.resources.master.limits.cpu }}
-              memory: {{ .Values.resources.master.limits.memory }}
-            requests:
-              cpu: {{ .Values.resources.master.requests.cpu }}
-              memory: {{ .Values.resources.master.requests.memory }}
-          command: ["/entrypoint.sh"]
-          args: ["master-only", "--no-format"]
-          envFrom:
-          - configMapRef:
-              name: alluxio-config
-          ports:
-          - containerPort: 19998
-            name: rpc
-          - containerPort: 19999
-            name: web
-          volumeMounts:
-            - name: alluxio-journal
-              mountPath: /journal
-            {{- if .Values.volumes.master.metastore }}
-            - name: alluxio-metastore
-              mountPath: {{ .Values.volumes.master.metastore.mountPath }}
-            {{- end }}
-            {{- if .Values.secrets }}
-            {{- if .Values.secrets.master }}
-            {{- range $key, $val := .Values.secrets.master }}
-            - name: secret-{{ $key }}-volume
-              mountPath: /secrets/{{ $val }}
-              readOnly: true
-            {{- end }}
-            {{- end }}
-            {{- end }}
-        - name: alluxio-job-master
-          image: {{ .Values.image }}:{{ .Values.imageTag }}
-          imagePullPolicy: {{ .Values.imagePullPolicy }}
-          resources:
-            limits:
-              cpu: {{ .Values.resources.jobMaster.limits.cpu }}
-              memory: {{ .Values.resources.jobMaster.limits.memory }}
-            requests:
-              cpu: {{ .Values.resources.jobMaster.requests.cpu }}
-              memory: {{ .Values.resources.jobMaster.requests.memory }}
-          command: ["/entrypoint.sh"]
-          args: ["job-master"]
-          envFrom:
-          - configMapRef:
-              name: alluxio-config
-          ports:
-          - containerPort: 20001
-            name: job-rpc
-          - containerPort: 20002
-            name: job-web
-          {{- if .Values.secrets }}
-          {{- if .Values.secrets.master }}
-          volumeMounts:
-            {{- range $key, $val := .Values.secrets.master }}
-            - name: secret-{{ $key }}-volume
-              mountPath: /secrets/{{ $val }}
-              readOnly: true
-            {{- end }}
-          {{- end }}
-          {{- end }}
-      restartPolicy: Always
-      volumes:
+      - name: alluxio-master
+        image: {{ $root.Values.image }}:{{ $root.Values.imageTag }}
+        imagePullPolicy: {{ $root.Values.imagePullPolicy }}
+        resources:
+          limits:
+            cpu: {{ $root.Values.resources.master.limits.cpu }}
+            memory: {{ $root.Values.resources.master.limits.memory }}
+          requests:
+            cpu: {{ $root.Values.resources.master.requests.cpu }}
+            memory: {{ $root.Values.resources.master.requests.memory }}
+        command: ["/entrypoint.sh"]
+        args: ["master-only", "--no-format"]
```

Comment:
embedded journal and ufs journal have different behavior. ufs journal formats if not formatted , i.e. detects if format is needed; whereas embedded does not. we may use init containers:
- https://github.com/cheyang/charts/blob/add_alluxio_chart/stable/alluxio/templates/master/statefulset.yaml#L45
- https://kubernetes.io/docs/concepts/workloads/pods/init-containers/

we can address this in a separate PR to pull in yang's improvements